single_kernel_mongo/state/charm_state.py

```diff
@@ -725,7 +730,7 @@ def is_shard_added_to_cluster(self) -> bool:
             ):
                 return False
             raise
-        except ServerSelectionTimeoutError:
+        except (ServerSelectionTimeoutError, AutoReconnect, NotPrimaryError):
```

Comment:
We need to backport this to 6/edge

Reply:
It's a cheap enough change, that this is our chance to squeeze this into the new stable we're working towards